pages/blog/blog-post.js

```diff
@@ -0,0 +1,301 @@
+  FacebookShareButton,
+} from 'react-share'
+import FaFacebook from 'react-icons/lib/fa/facebook'
+import FaTwitter from 'react-icons/lib/fa/twitter'
+
+import withLayout from '../../components/withLayout'
+import Head from '../../components/Head'
+import { getLocaleFromQuery } from '../../utils/locale'
+import { fetchBlogPost, fetchBlogPostPage } from '../../api/blog'
+import PageSection from '../../components/PageSection'
+import TeamMember from '../../components/TeamMember'
+import teamMemberShape from '../../propTypes/teamMember'
+import imageShape from '../../propTypes/image'
+import Hero from '../../components/Hero'
+import Markdown from '../../components/Markdown'
+import { smBreakpoint, mdBreakpoint, lgBreakpoint } from '../../styling/breakpoints'
+import theme from '../../styling/theme'
+import { rem } from '../../styling/typography'
+import Button from '../../components/Button'
+import Link from '../../components/Link'
+import { RouteNames as routes } from '../../routes'
+
+const BlogPostError = props => (
+  <div>
+    <Head title="FIXME: Oops" description="FIXME: Something went wrong" />
+    <PageSection>
+      <h1>FIXME: Oops</h1>
+      <p className="offset-lg-2 col-lg-8">{props.error}</p>
+    </PageSection>
+  </div>
+)
+
+BlogPostError.propTypes = {
+  error: PropTypes.string,
+}
+
+BlogPostError.defaultProps = {
+  error: null,
+}
+
+const BlogTitle = styled.h2`
+  margin-left: 0;
+  margin-right: 0;
+  text-align: left;
+  width: 100%;
+  color: ${theme.black};
+`
+
+const BlogMarkdown = styled(Markdown)`
+  p:first-child {
+    font-size: ${rem('24px')}
+  }
+
+  p img {
+    width: 100%;
+  }
+
+  @media(min-width:${smBreakpoint}) {
+    p img {
+      margin: 0 -25px;
+      width: calc(100% + 50px);
+    }
+  }
+
+  @media(min-width:${mdBreakpoint}) {
+    p img {
+      margin: 0 -30px;
+      width: calc(100% + 60px);
+    }
+  }
+
+  @media(min-width:${lgBreakpoint}) {
+    p img {
+      margin: 0 -120px;
+      width: calc(100% + 240px);
+    }
+  }
+`
+
+const TeamMemberAuthor = styled(TeamMember)`
+  justify-content: flex-start;
+`
+
+const DateContainer = styled.div`
+  margin-bottom: 1.5rem;
+`
+
+const AuthorContainer = styled.div`
+  margin-top: 1.5rem;
+`
+
+const ShareContainer = styled.div`
+  margin-top: 1.5rem;
+  display: flex;
+  flex-direction: column;
+  align-items: flex-end;
+`
+
+const ShareButtonRow = styled.div`
+  display: flex;
+  flex-direction: row;
+
+  > * {
+    padding: 10px;
+  }
+
+  > :last-child {
+    margin-right: -10px;
+  }
+`
+
+const SocialLink = styled.div`
+  color: #444444;
+  opacity: 0.6;
+  cursor: pointer;
+
+  &:hover {
+    color: #444444 !important;
+    opacity: 1;
+  }
+`
+
+const HorizontalRule = styled.hr`
+  border-width: 2px;
+  border-color: ${props => props.theme.pineCone};
+  margin-top: 4rem;
+`
+
+const ButtonContainer = styled.div`
+ display: flex;
+ flex-direction: row;
+ justify-content: space-between;
+`
+
+const BlogPostContent = (props) => {
```

Comment:
This component can be put into its own file.

Reply:
It could, but as far as I'm concerned these are just private implementation details of BlogPost so I don't really want to export them.